odoo/addons/test_lint/tests/_odoo_checker_gettext.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+import astroid
+from pylint import checkers, interfaces
+
+
+class OdooBaseChecker(checkers.BaseChecker):
+    __implements__ = interfaces.IAstroidChecker
+    name = 'odoo'
+
+    msgs = {
+        'E8502': (
+            'Bad usage of _ method.',
+            'gettext-variable',
+            'See https://www.odoo.com/documentation/13.0/reference/translations.html#variables'
+        )
+    }
+
+    def _get_func_name(self, node):
+        func_name = (
+            isinstance(node.func, astroid.Name) and node.func.name or
+            isinstance(node.func, astroid.Attribute) and node.func.attrname)
+        return func_name
+
+    def _check_concatenation(self, node):
+        is_bin_op = False
+        if isinstance(node, astroid.BinOp) and node.op in ('%', '+'):
+            # execute("..." % self._table)
+            if (isinstance(node.right, astroid.Attribute) and
+                    not node.right.attrname.startswith('_')):
+                is_bin_op = True
+            if isinstance(node.right, astroid.Tuple):
+                for elt in node.right.elts:
+                    if (isinstance(elt, astroid.Call) and
+                            # Ignoring for this case: execute("..." % _foo(...))
+                            not self._get_func_name(elt.func).startswith('_')):
+                        is_bin_op = True
+
+        is_format = False
+        # execute("...".format(self._table, table=self._table))
+        if (isinstance(node, astroid.Call) and
+                self._get_func_name(node.func) == 'format'):
+            for keyword in node.keywords or []:
+                if (isinstance(keyword.value, astroid.Attribute) and
+                        not keyword.value.attrname.startswith('_')):
+                    is_format = True
+                    break
+            for argument in node.args or []:
+                if (isinstance(argument, astroid.Name) and not argument.name.startswith('_')):
+                    is_format = True
+                    break
+
+        return is_bin_op or is_format
+
+    def _check_variable_inside_gettext(self, node):
+        func_name = self._get_func_name(node)
+        if not (
+            isinstance(node, astroid.Call) and node.args and
+            func_name in ('_', '_lt')
+        ):
```

Comment:
`_get_func_name` will crash if `node` is not an `astroid.Call`...
```suggestion
        if not (
            isinstance(node, astroid.Call) and node.args and
            self._get_func_name(node) in ('_', '_lt')
        ):
```

Reply:
> `_get_func_name` will crash if node is not an `astroid.Call`...

It's only called from `visit_call` so that's axiomatic.